std/algebra/native/sw_bls12377/g1.go

```diff
@@ -405,7 +323,15 @@ func (P *G1Affine) constScalarMul(api frontend.API, Q G1Affine, s *big.Int) *G1A
 	s.Mod(s, cc.fr)
 	cc.phi1(api, &phiQ, &Q)
 
-	k := ecc.SplitScalar(s, cc.glvBasis)
+	var k [2]big.Int
+	// if s=0, assign dummy 1s to k[0] and k[1]
+	if s.BitLen() == 0 {
```

Comment:
In constant case we can just return infinity point already here without needing to record any constraints. And similarly for s == 1 we can return Q directly. And in this case I would move the check before we perform any computations a few lines up (cc.phi1 and s.Mod)

Reply:
Yes you're right. Will do.